.github/workflows/build-executor.yml

```diff
@@ -0,0 +1,140 @@
+              "draft": false,
+              "prerelease": false,
+              "target_commitish": "'$BRANCH_NAME'"
+            }' \
+            "https://api.github.com/repos/${{ github.repository }}/releases"
+
+      - name: Update version in Ballerina.toml
+        if: ${{ inputs.bal_central_environment == 'PROD' }}
+        id: increment_patch_version
+        run: |
+          CURRENT_VERSION=$(grep -Po -m 1 '(?<=version = ")[\d.]+' ${{ inputs.working_dir }}/Ballerina.toml)
+          IFS='.' read -ra VERSION_PARTS <<< "$CURRENT_VERSION"
+          PATCH_VERSION=$((VERSION_PARTS[2] + 1))
+          NEW_VERSION="${VERSION_PARTS[0]}.${VERSION_PARTS[1]}.$PATCH_VERSION"
+          sed -i "0,/version = \"${CURRENT_VERSION}\"/s//version = \"${NEW_VERSION}\"/" ${{ inputs.working_dir }}/Ballerina.toml
+          echo "NEW_VERSION=${NEW_VERSION}" >> $GITHUB_OUTPUT
+          echo "CURRENT_VERSION=${CURRENT_VERSION}" >> $GITHUB_OUTPUT
+
+      - name: Ballerina Build after incrementing version
+        if: ${{ inputs.bal_central_environment == 'PROD' }}
+        uses: ballerina-platform/ballerina-action@2201.7.0
+        with:
+          args:
+            pack
+        env:
+          WORKING_DIR: ${{ inputs.working_dir }}
+          JAVA_HOME: /usr/lib/jvm/default-jvm
+
+      - name: Commit changes and make a PR
+        if: ${{ inputs.bal_central_environment == 'PROD' }}
+        run: |
+          # Extract the package name from working directory 
+          packageName=$(basename ${{ inputs.working_dir }})
+
+          # Commit changes
+          git config --global user.name ${{ secrets.BALLERINA_BOT_USERNAME }}
+          git config --global user.email ${{ secrets.BALLERINA_BOT_EMAIL }}
+          git add ${{ inputs.working_dir }}/Ballerina.toml
+          git add ${{ inputs.working_dir }}/Dependencies.toml    
+          git commit -m "[Release ${packageName} ${{ steps.increment_patch_version.outputs.CURRENT_VERSION }}] Prepare for next dev cycle"
+          git push origin ${{ steps.publish_release.outputs.BRANCH_NAME }}
+          
+          # Set the base and head branches
+          BASE_BRANCH="main"
+          HEAD_BRANCH="${{ steps.publish_release.outputs.BRANCH_NAME }}"
+          # Create the pull request using the GitHub REST API
+          RESPONSE=$(curl -s -X POST \
+            -H "Authorization: Bearer ${{ secrets.BALLERINA_BOT_TOKEN }}" \
+            -H "Accept: application/vnd.github.v3+json" \
+            -d '{
+              "title": "[Release ${packageName} ${{ steps.increment_patch_version.outputs.CURRENT_VERSION }}] Prepare for next dev cycle",
+              "body": "",
+              "head": "'"$HEAD_BRANCH"'",
+              "base": "'"$BASE_BRANCH"'"
+            }' \
+            "https://api.github.com/repos/${{ github.repository }}/pulls")
+          # Extract the pull request URL from the response
+          PR_URL=$(echo "$RESPONSE" | jq -r '.html_url')
+
+          echo "Pull Request created: $PR_URL"
```

Comment:
let's add a new line to indicate end of file(EoF). Let's do same to other files as well